backend/dataall/core/environment/db/environment_models.py

```diff
@@ -84,6 +84,14 @@ class ConsumptionRole(Base):
     groupUri = Column(String, nullable=False)
     IAMRoleName = Column(String, nullable=False)
     IAMRoleArn = Column(String, nullable=False)
+    dataallManaged = Column(Boolean, nullable=False, default=False)
     created = Column(DateTime, default=datetime.datetime.now)
     updated = Column(DateTime, onupdate=datetime.datetime.now)
     deleted = Column(DateTime)
+
+    def get_managed_share_policy_name(self):
+        return ConsumptionRole.generate_policy_name(self.environmentUri, self.IAMRoleName)
+
+    @staticmethod
+    def generate_policy_name(environment_uri, role_name):
```

Comment:
Maybe we should prefix with `dataall-env-...` or with `{resource_prefix}-env-...` similar to how we do for other resources created in an environment?

Reply:
it's already so

Reply:
It is currently defined as `return f'dataall-env-{environment_uri}-share-{iam_role_name}'` but maybe it should be using the resource_prefix instead of hardcoding dataall